src/inspect.rs

```diff
@@ -108,137 +292,136 @@ pub fn handle_inspect(inspect: &Inspect) -> Result<()> {
+            .expect("Incorrect block index")
+            .clone()
+            .map(hex::encode);
+        blocks.push(TokenBlock {
+            code: biscuit.print_block_source(i)?,
+            external_key,
+            revocation_id: revocation_ids
+                .get(i)
+                .map(hex::encode)
+                .unwrap_or_else(|| "n/a".to_owned()),
+        });
     }
 
+    let token_description = TokenDescription {
+        sealed: is_sealed,
+        root_key_id: biscuit.root_key_id(),
+        blocks,
+    };
+
+    let signatures_check;
+    let auth_result;
+    let query_result;
+
     if let Some(key_from) = public_key_from {
         let key = read_public_key_from(&key_from)?;
         let sig_result = biscuit.check_signature(|_| key);
-        if sig_result.is_err() {
-            println!("❌ Public key check failed 🔑");
-        }
-        let biscuit = sig_result?;
-        println!("✅ Public key check succeeded 🔑");
-
-        let mut authorizer_builder = biscuit.authorizer()?;
-        if let Some(auth_from) = authorizer_from {
-            read_authorizer_from(
-                &auth_from,
-                &inspect.param_arg.param,
-                &mut authorizer_builder,
-            )?;
-            if inspect.authorization_args.include_time {
-                let now = Utc::now().to_rfc3339();
-                let time_fact = format!("time({})", now);
-                authorizer_builder.add_fact(time_fact.as_ref())?;
-            }
-            let (_, _, _, policies) = authorizer_builder.dump();
-
-            let authorizer_result = authorizer_builder.authorize_with_limits(RunLimits {
-                max_facts: inspect
-                    .authorization_args
-                    .max_facts
-                    .unwrap_or_else(|| RunLimits::default().max_facts),
-                max_iterations: inspect
-                    .authorization_args
-                    .max_iterations
-                    .unwrap_or_else(|| RunLimits::default().max_iterations),
-                max_time: inspect
-                    .authorization_args
-                    .max_time
-                    .map_or_else(|| RunLimits::default().max_time, |d| d.to_std().unwrap()),
-            });
-
-            match authorizer_result {
-                Ok(i) => {
-                    println!("✅ Authorizer check succeeded 🛡️");
-                    println!(
-                        "Matched allow policy: {}",
-                        policies.get(i).expect("Incorrect policy index")
-                    );
-                }
+        signatures_check = Some(sig_result.is_ok());
 
-                Err(e) => {
-                    println!("❌ Authorizer check failed 🛡️");
-                    match e {
-                        Token::FailedLogic(l) => display_logic_error(&policies, &l),
-                        Token::RunLimit(l) => display_run_limit(&l),
-                        _ => {}
+        if let Ok(biscuit) = sig_result {
```

Comment:
Personal opinion, but wouldn't it be easier to read with a match block ?

Reply:
for huge blocks like this, i like if-let, as it indents things less. I do think the if/else semantics maps better to what we are actually doing, especially since we're performing side effects.